modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryRelatedObjectsResourceTest.java

```diff
@@ -1072,19 +1153,38 @@ private void _testPostCustomObjectEntryWithNestedSystemObjectEntry(
 				"code"
 			));
 
+		JSONObject systemObjectEntryJSONObject = HTTPTestUtil.invoke(
+			null,
+			StringBundler.concat(
+				jaxRsApplicationDescriptor.getRESTContextPath(),
+				StringPool.SLASH,
+				_getSystemObjectEntryId(
+					jsonObject.getString("id"), manyToOne, objectRelationship)),
+			Http.Method.GET);
+
+		_assertSystemObjectEntryValue(
```

Comment:
I think this way of validating the system object is a bit weird as you are:

- Validating the emailAddress field of the user account against the emailAddress returned by the GET endpoint
- Validating the custom object field of the user account against the custom field returned by the GET endpoint

What would happen if the related system object entry was not returned in the POST call?

I think we should verify that.

Apart from that, maybe we should modify the `_assertSystemObjectEntryValue` to check the system object in one invocation, not two